cores/esp32/HardwareSerial.h

```diff
@@ -61,7 +61,8 @@ typedef enum {
     UART_BUFFER_FULL_ERROR,
     UART_FIFO_OVF_ERROR,
     UART_FRAME_ERROR,
-    UART_PARITY_ERROR
+    UART_PARITY_ERROR,
+    UART_NO_ERROR
```

Comment:
I think that having `UART_NO_ERROR` be the first item in the enum makes most sense. It will equal to 0, which conforms to most/all error uses in C/C++ (and ESP-IDF)

Reply:
Done.